### PR TITLE
Switching out AppVeyor for GH workflow

### PR DESCRIPTION
**Pull request by srprash**
Since we have tested GH workflows for continuous builds on our other SDKs and it has worked out well for us, it's probably a good idea that we make the switch here as well to keep it consistent. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by srprash** [comment]
Will need to remove the AppVeyor application from the repo as well.

File: .github/workflows/pr-build.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+                
+        steps:
+        - name: Checkout repository
+        - uses: actions/checkout@v2
+          
+        - name: Install dependencies
+          run: dotnet restore sdk/AWSXRayRecorder.sln --locked-mode
+          
+        - name: Build solution
+          run: dotnet build sdk/AWSXRayRecorder.sln --no-restore
```
Comment (lupengamzn) [inline .github/workflows/pr-build.yml:27]: It's better for us to build package and test the dlls in `Release` configuration as it is the final stage that we [provided](https://github.com/aws/aws-xray-sdk-dotnet/blob/master/buildtools/build.proj#L7) for end users

File: .github/workflows/pr-build.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+          
+        - name: Run tests
+          run: dotnet test sdk/test/UnitTests/bin/**/${{matrix.version}}/AWSXRayRecorder.UnitTests.dll
```
Comment (lupengamzn) [inline .github/workflows/pr-build.yml:30]: Same here

File: .github/workflows/pr-build.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+            
+jobs:
+    build:
+        name: build
+        runs-on: ${{ matrix.os }}
+        
+        strategy:
+            fail-fast: false
+            matrix:
+                os: [windows-latest]
```
Comment (lupengamzn) [inline .github/workflows/pr-build.yml:16]: Do we need unit test for .net core on `linux`  as well?

Comment (srprash) [reply .github/workflows/pr-build.yml]: Yup. We should add linux as well. Good call.

File: .github/workflows/pr-build.yml (hunk 1 above)
Comment (srprash) [reply .github/workflows/pr-build.yml]: Yup

**Review by lupengamzn — APPROVED** [review]
LGTM

**Pull request merged**